content/200-concepts/300-more/400-comparisons/04-prisma-and-drizzle.mdx

```diff
@@ -0,0 +1,450 @@
+
+<ParallelBlocks>
+
+<block content="Prisma">
+
+```prisma
+model User {
+  id    Int     @id @default(autoincrement())
+  name  String?
+  email String  @unique
+  posts Post[]
+}
+
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  content   String?
+  published Boolean @default(false)
+  authorId  Int?
+  author    User?   @relation(fields: [authorId], references: [id])
+}
+```
+
+</block>
+
+<block content="Drizzle">
+
+```ts
+import {
+  boolean,
+  integer,
+  pgTable,
+  serial,
+  text,
+  uniqueIndex,
+  varchar,
+} from 'drizzle-orm/pg-core'
+
+export const users = pgTable('users', {
+  id: serial('id').primaryKey(),
+  name: varchar('name', { length: 256 }),
+  email: varchar('email', { length: 256 }).unique(),
+})
+
+export const posts = pgTable('posts', {
+  id: serial('id').primaryKey(),
+  title: varchar('title', { length: 256 }).notNull(),
+  content: text('content'),
+  published: boolean('published'),
+  authorId: integer('author_id').references(() => users.id),
+})
+```
+
+</block>
+
+</ParallelBlocks>
+
+### Migrations
+
+Migrations work similarly between Drizzle and Prisma. Both tools follow the approach of generating SQL files based on the provided model definitions and provide a CLI to execute them against the database. The SQL files can be modified before the migrations are executed so that any custom database operation can be performed with either migration system.
+
+### Querying
+
+Select queries are natural to construct in both Drizzle and Prisma. Using Drizzle’s Queries API, the two approaches are very similar:
+
+<ParallelBlocks>
+
+<block content="Prisma">
+
+```ts
+// find all users
+const allUsers = await prisma.user.findMany()
+
+// find a single user
+const user = await prisma.user.findFirst({
+  where: { id: 27 },
+})
+
+// find a unique user
+const user = await prisma.user.findUnique({
+  where: { email: 'nilu@prisma.io' },
+})
+```
+
+</block>
+
+<block content="Drizzle">
+
+```ts
+import { eq } from 'drizzle-orm'
+
+// find all users
+const allUsers = await db.query.users.findMany()
+
+// find a single user
+const user = await db.query.users.findFirst({
+  where: eq(users.id, 1),
+})
+
+// find a unique post
+const user = await db.query.users.findFirst({
+  where: eq(users.email, 'nilu@prisma.io'),
+})
+```
+
+</block>
+
+</ParallelBlocks>
+
+When performing a mutation, a `create`, `update`, or `delete`, the Drizzle Queries API is not available. In these cases, you will need to use Drizzle’s SQL-like APIs.
+
+<ParallelBlocks>
+
+<block content="Prisma">
+
+```ts
+// create a user
+const user = await prisma.user.create({
+  data: {
+    name: 'Nilu',
+    email: 'nilu@prisma.io',
+  },
+})
+
+// update a user
+const user = await prisma.user.update({
+  where: { email: 'nilu@prisma.io' },
+  data: { name: 'Another Nilu' },
+})
+
+// delete a user
+const deletedUser = await prisma.user.delete({
+  where: { email: 'nilu@prisma.io' },
+})
```

Comment:
```suggestion
const user = await prisma.user.create({
  data: {
    name: 'Prismo',
    email: 'prismo@prisma.io',
  },
})

// update a user
const user = await prisma.user.update({
  where: { email: 'prismo@prisma.io' },
  data: { name: 'Another Prismo' },
})

// delete a user
const deletedUser = await prisma.user.delete({
  where: { email: 'prismo@prisma.io' },
})
```

Is having Nilu in here something we usually do? 😄

Reply:
actually yes 🙃 we had decided that a while ago when we wanted to use diverse names for the user records in sample apps. we're also using that in all our examples: https://github.com/prisma/prisma-examples/blob/latest/typescript/graphql/prisma/seed.ts

we're only using first names though, so no personal info is disclosed or can be traced. would you prefer changing the approach?